indico/modules/rb/client/js/common/rooms/RoomEditModal.jsx

```diff
@@ -8,951 +8,200 @@
-  };
-
-  handleSubmit = async (data, form) => {
-    let {roomId} = this.props;
+    return camelizeKeys(response.data);
+  }, []);
+
+  const fetchRoomData = useCallback(async () => {
+    const resp = await Promise.all([
+      fetch(roomURL, {room_id: roomId}),
+      fetch(fetchRoomAttributesURL, {room_id: roomId}),
+      fetch(fetchRoomAvailabilityURL, {room_id: roomId}),
+    ]);
+    [setRoomDetails, setRoomAttributes, setRoomAvailability].forEach((x, i) => x(resp[i]));
+  }, [fetch, roomId]);
+
+  const handleSubmit = async (data, form) => {
     const changedValues = getChangedValues(data, form);
-    const basicDetailsKeys = [
-      'attributes',
-      'bookableHours',
-      'nonbookablePeriods',
-      'availableEquipment',
-    ];
-    const basicDetails = _.omit(changedValues, basicDetailsKeys);
-    const {availableEquipment, nonbookablePeriods, bookableHours, attributes} = changedValues;
-    let submitState = 'success';
-    let submitError;
+    const {
+      attributes,
+      bookableHours,
+      nonbookablePeriods,
+      availableEquipment,
+      ...basicDetails
+    } = changedValues;
     try {
-      if (this.newRoom) {
-        roomId = await this.createRoom(basicDetails);
-        this.setState({newRoomId: roomId});
-      } else {
-        await this.saveBasicDetails(roomId, basicDetails);
+      if (isNewRoom()) {
+        const payload = snakifyKeys(basicDetails);
+        payload.location_id = locationId;
+        const response = await indicoAxios.post(roomsURL(), payload);
+        setNewRoomId(response.data.id);
+      } else if (!_.isEmpty(basicDetails)) {
+        await indicoAxios.patch(roomURL({room_id: roomId}), snakifyKeys(basicDetails));
+      }
+      if (availableEquipment) {
+        await indicoAxios.post(
+          updateRoomEquipmentURL({room_id: roomId}),
+          snakifyKeys({available_equipment: availableEquipment})
+        );
+      }
+      if (attributes) {
+        await indicoAxios.post(updateRoomAttributesURL({room_id: roomId}), {attributes});
+      }
+      if (bookableHours || nonbookablePeriods) {
+        const availability = {bookableHours, nonbookablePeriods};
+        await indicoAxios.post(
+          updateRoomAvailabilityURL({room_id: roomId}),
+          snakifyKeys(availability)
+        );
       }
-      await this.saveEquipment(roomId, availableEquipment);
-      await this.saveAttributes(roomId, attributes);
-      await this.saveAvailability(roomId, changedValues, nonbookablePeriods, bookableHours);
-
       // reload room so the form gets new initialValues
-      await this.fetchRoomData(roomId);
-      this.setState({
-        wasEverSaved: true,
-        submitState,
-      });
+      if (!isNewRoom()) {
+        setWasEverUpdated(true);
+        fetchRoomData();
+      }
     } catch (e) {
-      submitError = handleSubmitError(e);
-      submitState = 'error';
-      this.setState({submitState});
+      return camelizeKeys(handleSubmitError(e));
```

Comment:
I'd rather keep the `snake_case` names on the various form fields to avoid having to snakify/camelize here.

Reply:
And convert all the input names into `snake_case`?

Reply:
Yeah, in this particular case I'd do it.

I think camelize for responses coming back from the server still makes sense in many cases, but for form fields I'd rather use the same names - it's nice if you can search for the name of a field and find both the server-side and client-side code for it!

Reply:
Yes, that is helpful. I've updated it. Even though having to convert from snake case to camel case in between code and input names can become annoying e.g.
from:
`const {availableEquipment, availableSomething} = camel(input)`
to:
`const {available_equiment: availableEquipment, available_something: availableSomething} = input`

However, I think for this case it is still manageable.

Reply:
Indeed this is a bit ugly... Well you can always use e.g. `input.available_equipment` depending on how you are using things. We don't have to destructure everywhere - it's a convenient thing to do in many places, but I think it's best to decide case-by-case whether it's the right / more readable thing to do or not.